x-pack/plugins/dashboard_enhanced/public/services/drilldowns/actions/flyout_create_drilldown/flyout_create_drilldown.test.tsx

```diff
@@ -5,164 +5,152 @@
+    dynamicActions: new DynamicActionManager({
+      storage: new MemoryActionStorage(),
+      isCompatible: async () => true,
+      uiActions: uiActionsEnhancedPluginMock.createStartContract(),
+    }),
+  },
+  parentApi: {
+    type: 'dashboard',
+  },
+  supportedTriggers: () => {
+    return ['VALUE_CLICK_TRIGGER'];
+  },
+  viewMode: new BehaviorSubject<ViewMode>('edit'),
+};
 
 test('title is a string', () => {
-  expect(typeof new FlyoutCreateDrilldownAction(actionParams).getDisplayName() === 'string').toBe(
-    true
-  );
+  expect(typeof createAction().getDisplayName() === 'string').toBe(true);
 });
 
 test('icon exists', () => {
-  expect(typeof new FlyoutCreateDrilldownAction(actionParams).getIconType() === 'string').toBe(
-    true
-  );
+  expect(typeof createAction().getIconType() === 'string').toBe(true);
 });
 
-interface CompatibilityParams {
-  isEdit?: boolean;
-  isValueClickTriggerSupported?: boolean;
-  isEmbeddableEnhanced?: boolean;
-  rootType?: string;
-  actionFactoriesTriggers?: string[];
-}
-
 describe('isCompatible', () => {
-  const drilldownAction = new FlyoutCreateDrilldownAction(actionParams);
-
-  async function assertCompatibility(
-    {
-      isEdit = true,
-      isValueClickTriggerSupported = true,
-      isEmbeddableEnhanced = true,
-      rootType = 'dashboard',
-      actionFactoriesTriggers = ['VALUE_CLICK_TRIGGER'],
-    }: CompatibilityParams,
-    expectedResult: boolean = true
-  ): Promise<void> {
-    uiActionsEnhanced.getActionFactories.mockImplementation(() => [
-      {
-        supportedTriggers: () => actionFactoriesTriggers,
-        isCompatibleLicense: () => true,
-      } as unknown as UiActionsEnhancedActionFactory,
-    ]);
-
-    let embeddable = new MockEmbeddable(
-      { id: '', viewMode: isEdit ? ViewMode.EDIT : ViewMode.VIEW },
-      {
-        supportedTriggers: isValueClickTriggerSupported ? ['VALUE_CLICK_TRIGGER'] : [],
-      }
-    );
-
-    embeddable.rootType = rootType;
-
-    if (isEmbeddableEnhanced) {
-      embeddable = enhanceEmbeddable(embeddable);
-    }
-
-    const result = await drilldownAction.isCompatible({
-      embeddable,
-    });
-
-    expect(result).toBe(expectedResult);
-  }
-
-  const assertNonCompatibility = (params: CompatibilityParams) =>
-    assertCompatibility(params, false);
-
   test("compatible if dynamicUiActions enabled, 'VALUE_CLICK_TRIGGER' is supported, in edit mode", async () => {
-    await assertCompatibility({});
+    const action = createAction();
+    expect(await action.isCompatible({ embeddable: compatibleEmbeddableApi })).toBe(true);
   });
 
   test('not compatible if embeddable is not enhanced', async () => {
-    await assertNonCompatibility({
-      isEmbeddableEnhanced: false,
-    });
+    const action = createAction();
+    const embeddableApi = {
+      ...compatibleEmbeddableApi,
+      enhancements: undefined,
+    };
+    expect(await action.isCompatible({ embeddable: embeddableApi })).toBe(false);
```

Comment:
The simplicity and readability of these tests really shows off why we're moving into composition for Embeddables, very nice!